pep-0622.rst

```diff
@@ -1621,11 +1615,37 @@ the below match would generate the following call::
       False,
   )
 
-This would allow faster ``__match__()`` implementations and give better
-support for customization in user-defined classes. There is however a big
-downside to this: it would make the basic implementation of this method quite
-complicated. Also, there would be a performance penalty if the user did not
-treat the pattern object properly.
+One drawback of this protocol is that the arguments to ``__match__``
+would be expensive to construct, and could not be pre-computed due to
+the fact that, because of the way names are bound, there are no real
+constants in Python. It also meant that the ``__match__`` method would
+have to re-implement much of the logic of matching which would otherwise
+be implemented in C code in the Python VM. As a result, this option would
+perform poorly compared to an equilvalent ``if``-statement.
+
+The simpler protocol suffered from the fact that although it was more
+performance, it was much less flexible, and did not allow for many of
```

Comment:
```suggestion
performant, it was much less flexible, and did not allow for many of
```

Reply:
So apparently there is some controversy as to whether 'performant' is a word. I'll change it.